compiler/rustc_lint/src/types.rs

```diff
@@ -572,32 +572,35 @@ fn lint_nan<'tcx>(
     }
 
     fn eq_ne(
+        cx: &LateContext<'_>,
         e: &hir::Expr<'_>,
         l: &hir::Expr<'_>,
         r: &hir::Expr<'_>,
         f: impl FnOnce(Span, Span) -> InvalidNanComparisonsSuggestion,
     ) -> InvalidNanComparisons {
-        let suggestion =
+        let suggestion = (!cx.tcx.hir().is_inside_const_context(e.hir_id)).then(|| {
```

Comment:
nit: this is a somewhat confusing way of avoiding an `if` statement :sweat_smile: